### PR TITLE
Fix the way failed and success bump tests are counted and fix statistics

### PR DESCRIPTION
**Pull request by joaquinhernandezg**
According to what is reported in [SITCOM-1231](https://jira.lsstcorp.org/secure/RapidBoard.jspa?rapidView=275&view=detail&selectedIssue=SITCOM-1231), the following changes are introduced:

- The introduction of the function `find_individual_bump_tests`, which finds the individual bump tests associated with a given actuator in the `forceActuatorBumpTestStatus` telemetry. The main idea of this function is to find all sequences starting in NOTTESTED and finishing in PASSED or FAILED.  The algorithm is the following:
    - For a given actuator
    - Find all the NOTTESTED events whose next state is different than NOTTESTED. This happens when an actuator starts a bump test.
    - The testing sequence is "TESTINGPOSITIVE"->"TESTINGPOSITIVEWAIT"-> "TESTINGNEGATIVE"->"TESTINGNEGATIVEWAIT". Find all sequences (by truncating the testing sequence) that end in "FAILED". (e.g. NOTTESTED->TESTINGPOSITIVE->TESTINGPOSITIVEWAIT->FAILDED ). These are FAILED tests.
    - Save a list of DF containing the failed testing sequences. (one df per test sequence)
    - Find all ending in "PASSED" (NOTTESTED-> TESTINGPOSITIVE->TESTINGPOSITIVEWAIT-> TESTINGNEGATIVE->TESTINGNEGATIVEWAIT-> PASSED)
    - Save a list of DF containing the passed testing sequence.

- Change the `calculate_failures` function:
    - Now count the passed and failed failures for primary and secondary actuators.
    - Now the failure ratio is estimated as FAILURES_i/TESTS_i  where FAILURES_i and TESTS_i is the total number of failures for the actuator i, and the total number of tests, respectively. (Previously it was FAILURES_i/sum(FAILURES_i))

- Change the `plot_stacked_failures` and `create_3d_heatmap` to use the new statistics.

### CODE REVIEW

File: notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOMTN-083_m1m3_bump_test_failure_analysis.ipynb [hunk 1]
```diff
@@ -502,10 +502,150 @@
     "    return df\n",
```
Comment (nsevilla) [inline notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOMTN-083_m1m3_bump_test_failure_analysis.ipynb:502]: <pre class="ql-syntax" spellcheck="false">Line #271.        # Calculate the total number of failures
</pre><p>I think this should say total number of tests</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/117/discussion/#comment-1506766636'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (nsevilla) [inline notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOMTN-083_m1m3_bump_test_failure_analysis.ipynb:502]: <pre class="ql-syntax" spellcheck="false">Line #32.        # Calculate the total number of failures
</pre><p>Same comment, should it say 'total number of tests'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/notebooks_vandv/pull/117/discussion/#comment-1506766638'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

**Review by nsevilla — APPROVED** [review]
It took me a bit but I think I understood what you did, looks good. Just a very minor change in a couple of comments (I think I am correct, disregard if not).

**Comment by joaquinhernandezg** [comment]
Made the change and Squashed the commit with the previous one, as it just a typo.

**Pull request merged**